sw/airborne/mcu_periph/uart.h

```diff
@@ -36,8 +36,13 @@
 #define UART_RX_BUFFER_SIZE 128
 #endif
 #ifndef UART_TX_BUFFER_SIZE
+#ifdef STM32F4  //the F4 has enough memory, and the PX4 bootloader needs more then 128
```

Comment:
As you want.. if you move it to uart_arch.h, you might as well rais the rx buffer size to the same value for F4.